repair/repair.cc

```diff
@@ -796,17 +796,22 @@ static future<> do_repair_ranges(lw_shared_ptr<repair_info> ri) {
     co_return;
 }
 
-// repair_ranges repairs a list of token ranges, each assumed to be a token
+
+// Repairs a list of token ranges, each assumed to be a token
 // range for which this node holds a replica, and, importantly, each range
 // is assumed to be a indivisible in the sense that all the tokens in has the
 // same nodes as replicas.
-static future<> repair_ranges(lw_shared_ptr<repair_info> ri) {
-    ri->rs.repair_module()->add_repair_info(ri->id.id, ri);
-    return do_repair_ranges(ri).then([ri] {
+future<> tasks::shard_repair_task_impl::run() {
+    if (_ex) {
+        return make_exception_future(_ex);
+    }
+
+    _ri->rs.repair_module()->add_repair_info(_ri->id.id, _ri);
+    return do_repair_ranges(_ri).then([ri = _ri] {
```

Comment:
_ri is kept alive as a member of the task.
I suggest to get a reference to it in the continuation body rather than capturing the shared ptr and capture `this` instead.